apps/src/lab2/views/DancerCanvas.tsx

```diff
@@ -169,15 +176,22 @@ const DancerCanvas: React.FC<Props> = ({
 
     const renderer = rendererRef.current;
     const totalFrames = renderer?.getDurationFrames?.() ?? 1;
-    const targetFps = 60;
 
+    // We aim to cycle through the entire animation each second.
+    // The animation speed should be similar to with a 120 bpm song.
+    const targetFps = 48;
+
+    let mirror = 1;
     const tick = (currentMs: number) => {
       const elapsedMs = currentMs - startMs;
       const frameIndex =
         Math.floor((elapsedMs * targetFps) / 1000) % totalFrames;
+      // Detect when the animation loops back to frame 0
+      if (frameRef.current === totalFrames - 1 && frameIndex === 0) {
+        mirror *= -1;
```

Comment:
Is there a chance we could miss this step if we are dropping frames on a slower device?  Could we make this an absolute calculation based on which frame we're up to, rather than a flip that happens on a specific frame?

Reply:
If we could possibly be on the same frame twice, we could accidentally flip twice, which is why I needed to compare. Maybe we could soften to just flip whenever the current frame is less than the previous frame (since we only count up)?